### ISSUE TITLE
linter: eslint-plugin-import(exports-last) incorrect report

### ISSUE THREAD

**Issue by ValentinGurkov** [oxc-project/oxc#10231]
### What version of Oxlint are you using?

0.16.4

### What command did you run?

oxlint --deny-warnings

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "plugins": ["typescript", "unicorn", "perf", "oxc", "import", "promise", "node", "vitest"],
    "categories": {
        "correctness": "warn",
        "suspicious": "warn",
        "pedantic": "off",
        "perf": "warn",
        "style": "warn",
        "restriction": "off",
        "nursery": "off"
    },
    "env": {
        "builtin": true,
        "es2025": true,
        "shared-node-browser": true
    },
    "ignorePatterns": [".next", "node_modules", "build"],
    "globals": {
        "Buffer": "readonly",
        "clearImmediate": "readonly",
        "global": "readonly",
        "process": "readonly",
        "setImmediate": "readonly",
        "afterAll": "readonly",
        "afterEach": "readonly",
        "assert": "readonly",
        "assertType": "readonly",
        "beforeAll": "readonly",
        "beforeEach": "readonly",
        "chai": "readonly",
        "describe": "readonly",
        "expect": "readonly",
        "expectTypeOf": "readonly",
        "it": "readonly",
        "onTestFailed": "readonly",
        "onTestFinished": "readonly",
        "suite": "readonly",
        "test": "readonly",
        "vi": "readonly",
        "vitest": "readonly"
    },
    "rules": {
        "yoda": "off",
        "no-ternary": "off",
        "no-var": "warn",
        "prefer-rest-params": "warn",
        "prefer-spread": "warn",
        "no-duplicate-imports": "off",
        "unicorn/no-null": "off",
        "unicorn/filename-case": "off",
        "sort-imports": "off",
        "sort-keys": "off",
        "no-magic-numbers": "off",
        "for-direction": "warn",
        "no-async-promise-executor": "warn",
        "no-case-declarations": "warn",
        "no-class-assign": "warn",
        "no-compare-neg-zero": "warn",
        "no-cond-assign": "warn",
        "no-const-assign": "warn",
        "no-constant-binary-expression": "warn",
        "no-constant-condition": "warn",
        "no-control-regex": "warn",
        "no-debugger": "warn",
        "no-delete-var": "warn",
        "no-dupe-class-members": "warn",
        "no-dupe-else-if": "warn",
        "no-dupe-keys": "warn",
        "no-duplicate-case": "warn",
        "no-empty": "warn",
        "no-empty-character-class": "warn",
        "no-empty-pattern": "warn",
        "no-empty-static-block": "warn",
        "no-ex-assign": "warn",
        "no-extra-boolean-cast": "warn",
        "no-fallthrough": "warn",
        "no-func-assign": "warn",
        "no-global-assign": "warn",
        "no-import-assign": "warn",
        "no-invalid-regexp": "warn",
        "no-irregular-whitespace": "warn",
        "no-loss-of-precision": "warn",
        "no-new-native-nonconstructor": "warn",
        "no-nonoctal-decimal-escape": "warn",
        "no-obj-calls": "warn",
        "no-prototype-builtins": "warn",
        "no-redeclare": "warn",
        "no-regex-spaces": "warn",
        "no-self-assign": "warn",
        "no-setter-return": "warn",
        "no-shadow-restricted-names": "warn",
        "no-sparse-arrays": "warn",
        "no-this-before-super": "warn",
        "no-unexpected-multiline": "warn",
        "no-unsafe-finally": "warn",
        "no-unsafe-negation": "warn",
        "no-unsafe-optional-chaining": "warn",
        "no-unused-labels": "warn",
        "no-unused-private-class-members": "warn",
        "no-unused-vars": "warn",
        "no-useless-catch": "warn",
        "no-useless-escape": "warn",
        "no-with": "warn",
        "require-yield": "warn",
        "use-isnan": "warn",
        "valid-typeof": "warn",
        "@typescript-eslint/ban-ts-comment": "warn",
        "no-array-constructor": "warn",
        "@typescript-eslint/no-duplicate-enum-values": "warn",
        "@typescript-eslint/no-empty-object-type": "warn",
        "@typescript-eslint/no-explicit-any": "warn",
        "@typescript-eslint/no-extra-non-null-assertion": "warn",
        "@typescript-eslint/no-misused-new": "warn",
        "@typescript-eslint/no-namespace": "warn",
        "@typescript-eslint/no-non-null-asserted-optional-chain": "warn",
        "@typescript-eslint/no-require-imports": "warn",
        "@typescript-eslint/no-this-alias": "warn",
        "@typescript-eslint/no-unnecessary-type-constraint": "warn",
        "@typescript-eslint/no-unsafe-declaration-merging": "warn",
        "@typescript-eslint/no-unsafe-function-type": "warn",
        "no-unused-expressions": "warn",
        "@typescript-eslint/no-wrapper-object-types": "warn",
        "no-throw-literal": "off",
        "@typescript-eslint/prefer-as-const": "warn",
        "@typescript-eslint/prefer-namespace-keyword": "warn",
        "prefer-promise-reject-errors": "off",
        "require-await": "off",
        "@typescript-eslint/triple-slash-reference": "warn",
        "@typescript-eslint/consistent-type-definitions": ["warn", "type"]
    }
}
```

### What happened?

I'm running oxlint on a codebase based on `shadcn` components and I believe `eslint-plugin-import(exports-last)` flags an error incorrectly.

```jsx
import { cn } from '@/lib/utils';
import { Slot } from '@radix-ui/react-slot';
import { type VariantProps, cva } from 'class-variance-authority';
import type { ButtonHTMLAttributes, ReactElement, Ref } from 'react';

export const buttonVariants = cva(
    'inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0',
    {
        variants: {
            variant: {
                default: 'bg-primary text-primary-foreground shadow hover:bg-primary/90',
                destructive: 'bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90',
                outline: 'border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground',
                secondary: 'bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80',
                ghost: 'hover:bg-accent hover:text-accent-foreground',
                link: 'text-primary underline-offset-4 hover:underline'
            },
            size: {
                default: 'h-9 px-4 py-2',
                sm: 'h-8 rounded-md px-3 text-xs',
                lg: 'h-10 rounded-md px-8',
                icon: 'h-9 w-9'
            }
        },
        defaultVariants: {
            variant: 'default',
            size: 'default'
        }
    }
);

export type ButtonProps = {
    asChild?: boolean;
} & ButtonHTMLAttributes<HTMLButtonElement> &
    VariantProps<typeof buttonVariants>;

export const Button = ({
    ref,
    className,
    variant,
    size,
    asChild = false,
    ...props
}: Readonly<ButtonProps & { ref?: Ref<HTMLButtonElement | null> }>): ReactElement => {
    const Comp = asChild ? Slot : 'button';
    return <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />;
};
Button.displayName = 'Button';
```

Due to `Button.displayName = 'Button';` being on the last line the linter reports an error.
There is an alternative way to declare the component that will not result in this error:
```jsx
export function Button({
    ref,
    className,
    variant,
    size,
    asChild = false,
    ...props
}: Readonly<ButtonProps & { ref?: Ref<HTMLButtonElement | null> }>): ReactElement {
    const Comp = asChild ? Slot : 'button';
    return <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />;
}
```

Though in my opinion, the current code should also not raise an error. Of course, you might disagree.

**Comment by huangtiandi1999** [comment]
1. The same code is also reported when using eslint-plugin-import, [the official rule definition](https://github.com/import-js/eslint-plugin-import/blob/91f809b28323bfbd27749bae84daed00511b07e5/docs/rules/exports-last.md?plain=1#L5):
> **This rule will report any export declarations that comes before any non-export statements.**

2. If the rule allowed code after export, its purpose (centralized exports) would be defeated.

**Comment by camc314** [comment]
thanks @huangtiandi1999 

could also write it as: which (i think) shouldn't error
```
const Button = ({
    ref,
    className,
    variant,
    size,
    asChild = false,
    ...props
}: Readonly<ButtonProps & { ref?: Ref<HTMLButtonElement | null> }>): ReactElement => {
    const Comp = asChild ? Slot : 'button';
    return <Comp className={cn(buttonVariants({ variant, size, className }))} ref={ref} {...props} />;
};
Button.displayName = 'Button';

export { Button };
```
